src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -136,6 +168,33 @@ public object Clone()
             return name;
         }
 
+        private static Func<string, AssemblyName>? s_getAssemblyName;
+        private static Func<string, AssemblyName> InitGetAssemblyName()
+        {
+            Type? readerType = Type.GetType(
+                    "System.Reflection.Metadata.MetadataReader, System.Reflection.Metadata, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
```

Comment:
Is there a reason why you are using `Version=4.0.1.0`? I think it should be either `6.0`; or you can skip the version altogether - it is what we are doing in `src\coreclr\System.Private.CoreLib\src\System\Reflection\Emit\ILGenerator.cs` for System.Console.

Reply:
I've just copied the pattern from another similar case for stack traces. I am not sure if the version is needed.